<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>then</title>
</head>
<body>
<script>
    const p = new Promise((resolve, reject) => {
        // resolve(1);
        reject(2);
    });
    p.then(
        (data) => {
            console.log('success1');
            console.log(data);
        },
        (err) => {
            console.log('err1');
            console.log(err);
            // 在then的回掉函数中，return后面的东西，会用promise包装一下，return的值就是下一个then里面的参数
            /*在没有返回值的时候，默认是 return undefined; 的
            // 等价于
            return new Promise((resolve, reject) => {
                resolve(undefined);
            });*/

            /*return 123;
            // 等价于
            return new Promise((resolve, reject)=>{
                resolve(123);
            });*/

            // 默认返回的永远都是成功状态的Promise对象
            // 如果想要失败状态下的Promise对象
            return new Promise((resolve, reject) => {
                reject('reason');
            })
        }
    ).then(
        (data) => {
            console.log('success2');
            console.log(data); // 打印123
        },
        (err) => {
            console.log('err2');
            console.log(err); // 打印reason
        }
    )
</script>
</body>
</html>